### PR TITLE
[AMORO-1893][Optimizer] Fix unit conversion for parameters `-msz` to spillmap configuration

### LINKED ISSUE #1893: [Bug]: Fix unit conversion for parameters `-msz` to spillmap configuration
### What happened?

The spillmap is inconsistent with the parameter (`-msz`) unit, resulting in the use of the disk map without reaching the configuration, which affects the `optimizer` performance.

### Affects Versions

master/0.5.0

### What engines are you seeing the problem on?

Optimizer

### How to reproduce

_No response_

### Relevant log output

_No response_

### Anything else

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by zhongqishang**
## Why are the changes needed?

Close #1893 .

## Brief change log

- Fix unit conversion for parameters `-msz` to spillmap configuration
- Add optimizer parameter doc

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / **no**)
- If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: docs/admin-guides/managing-optimizers.md [hunk 1]
```diff
@@ -123,15 +123,28 @@ Currently, only pptimizer scaled through the dashboard can be released on dashbo
...
+ -Dtaskmanager.memory.process.size=2048m \
+ -c com.netease.arctic.optimizer.flink.FlinkOptimizer \
+ ${ARCTIC_HOME}/plugin/optimize/OptimizeJob.jar \
+ -a 127.0.0.1:1261 \
+ -g flinkGroup \
+ -p 1 \
+ -eds \
+ -dsp /tmp \
+ -msz 512 \
+ -id 9c12edca-b0bf-434d-a954-dd8aab88a284
```
Comment (zhoujinsong) [inline docs/admin-guides/managing-optimizers.md:137]: External optimizer do not need this `id` parameter.

**Review by zhoujinsong — COMMENTED** [review]
@zhongqishang Thanks a lot for reporting and fixing this bug!

File: docs/admin-guides/managing-optimizers.md (hunk 1 above)
Comment (zhongqishang) [reply docs/admin-guides/managing-optimizers.md]: Done.

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**